src/Libraries/Microsoft.Extensions.DependencyInjection.AutoActivation/AutoActivationExtensions.Keyed.cs

```diff
@@ -0,0 +1,405 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.Shared.DiagnosticIds;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+public static partial class AutoActivationExtensions
+{
+    /// <summary>
+    /// Enforces singleton activation at startup time rather then at runtime.
+    /// </summary>
+    /// <typeparam name="TService">The type of the service to activate.</typeparam>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
+    /// <param name="serviceKey">The <see cref="ServiceDescriptor.ServiceKey"/> of the service.</param>
+    /// <returns>A reference to this instance after the operation has completed.</returns>
+    [Experimental(diagnosticId: Experiments.AutoActivation, UrlFormat = Experiments.UrlFormat)]
+    public static IServiceCollection ActivateKeyed<TService>(
```

Comment:
Should we rename it to `ActivateKeyedSingleton`? All DI-related extension method specify desired lifetime it their name

Reply:
Alternatively, if we don't care about lifetime, we should remove `singleton` from XML docs